### DISCUSSION TITLE
Any idea why my Metamorphisis icon in Hekili has a timer that loops?

### DISCUSSION THREAD

**Discussion by azurejim** [Hekili/hekili#1234] [General]
https://user-images.githubusercontent.com/50629059/133813959-55bb0dfd-feff-44d3-b86b-ec8b3f06a6f1.mp4

**Comment by Hekili** [comment]
You can pause and mouseover the icon to see what the conditions are for that icon to be shown.

My best guess would be that it's based on using Metamorphosis before a fight ends (i.e., default fight length is 30 until time-to-die calculations can be made, and the priority uses Metamorphosis is remaining fight time is less than 25 seconds).

Out of combat, the addon updates recommendations every second, which would cause that timer to go further out.

**Reply by azurejim** [reply]
![Picture1](https://user-images.githubusercontent.com/50629059/133829084-d7eb288d-7335-41d2-8680-e09efd23650b.png)
i haven't used meta in awhile.  when i pause the addon, it counts down all the way, but when i unpause it just keeps looping from 5 seconds to 4 over and over.  and you're right, it's after i'm done with a battle.

**Comment by Hekili** [comment]
You can see from the conditions displayed that my hunch was correct.  fight_remains defaults to 30 seconds.  In 5.01 seconds, Metamorphosis' conditions would be met, in this hypothetical fight that hasn't started.

It ticks down when paused because that "5.01 seconds from when the recommendations were generated" time is passing when it's paused.  When unpaused, the "5.01 seconds from now" is restarted every time the addon updates recommendations.

In any case, it's probably a good idea for these entries that are based on fight_remains to only apply to boss fights.  In that case, Metamorphosis wouldn't be shown right now at all.
